part2/Crypotology(Lab1)/src/main/java/lab1/ExtendedEuclidean.java

```diff
@@ -0,0 +1,40 @@
+package lab1;
+
+import java.math.BigInteger;
+import java.util.Random;
+
+// https://brilliant.org/wiki/extended-euclidean-algorithm/
+// https://en.wikibooks.org/wiki/Algorithm_Implementation/Mathematics/Extended_Euclidean_algorithm
+// https://en.wikipedia.org/wiki/Extended_Euclidean_algorithm
+// ax + by = gcd(a, b)
+public class ExtendedEuclidean {
+    // Assumes a and b are positive
+    public static BigInteger[] xgcd(BigInteger a, BigInteger b) {
+        BigInteger[] qrem;
+        BigInteger[] result = new BigInteger[3];
+        BigInteger x = a, y=b;
+        BigInteger x0 = BigInteger.ONE, x1 = BigInteger.ZERO;
+        BigInteger y0 = BigInteger.ZERO, y1 = BigInteger.ONE;
+        while (true){
+            qrem = x.divideAndRemainder(y); x = qrem[1];
```

Comment:
Please do not ignore Java Code Convention

Reply:
I agree about that.  Done.